refactor: simplify some cases of common.formatQueryValue

laumair opened the pull request:
For https://github.com/nock/nock/issues/1404 & https://github.com/nock/nock/issues/1285.

- Replaces [forOwn](https://lodash.com/docs/4.17.11#forOwn) with [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)
- Adds unit test coverage for [formatQueryValue](https://github.com/nock/nock/blob/beta/lib/common.js#L442)

paulmelnikow commented:
@laumair Thanks so much for this contribution! It's great to have more help on #1404!

@mastermatt Would you be up for giving this a look?

mastermatt commented on lib/common.js:
```diff
@@ -459,20 +459,22 @@ function formatQueryValue(key, value, stringFormattingFn) {
-        tmpObj[subPair[0]] = subPair[1]
-      })
-      value = tmpObj
+      value = Object.keys(value).reduce(function(acc, subKey) {
```
`Object.entries` is probably better suited here.

```js
Object.entries(value).reduce(function(acc, [subVal, subKey]) {
  const subPair = formatQueryValue(subKey, subVal, stringFormattingFn)
  acc[subPair[0]] = subPair[1]

  return acc
}, {})
```

laumair replied:
Fixed with https://github.com/nock/nock/pull/1598/commits/6a785a816dc1382572e6c7a67ac3a4e37378d880.

mastermatt commented on tests/test_common.js:
```diff
@@ -453,3 +453,103 @@ test('percentEncode encodes extra reserved characters', t => {
+  // when stringFormattingFn is not passed
+  t.deepEqual(common.formatQueryValue(0, 'foo'), [0, 'foo'])
+
+  // when stringFormattingFn is passed
```
We've been working on making the unit size of tests as small as possible, so I'd probably break this up into two separate tests.

laumair replied:
Addressed in https://github.com/nock/nock/pull/1598/commits/37196f142c211708611e5b6a6e1e33255acf5681

mastermatt commented on tests/test_common.js:
```diff
@@ -453,3 +453,103 @@ test('percentEncode encodes extra reserved characters', t => {
+    return typeof str === 'string' ? str.toLowerCase() : str
+  }
+
+  // when stringFormattingFn is passed
```
...same as above.

laumair replied:
Addressed in https://github.com/nock/nock/pull/1598/commits/37196f142c211708611e5b6a6e1e33255acf5681

laumair commented:
@mastermatt Thank you for the review. 

>  I think this would fall into a refactor instead of test category. 

I believe you are referring to the PR title because the branch name already has `refactor/`. I've also updated the PR title. 

I have also addressed your comments with https://github.com/nock/nock/pull/1598/commits/6a785a816dc1382572e6c7a67ac3a4e37378d880 & https://github.com/nock/nock/pull/1598/commits/37196f142c211708611e5b6a6e1e33255acf5681

mastermatt commented:
Travis is failing because of Prettier.
`npm run prettier`

laumair commented:
> Travis is failing because of Prettier.
> `npm run prettier`

Ah! Fixed with https://github.com/nock/nock/pull/1598/commits/6cffb896e14e1146c5351eac000b3c063b5bd518.

mastermatt commented:
The changes in `lib` look good to me.

I hesitate on approving the PR in general because of the tests and this comment https://github.com/nock/nock/pull/1588#discussion_r301230231

@paulmelnikow @gr2m  Should these tests be rewritten in terms of calls to `http`?

gr2m commented:
I’ve rebased on latest master. Looking at the changes right now

mastermatt commented:
@gr2m is your recommendation to remove the new tests in this PR and merge just the `lib/common.js` refactor?

gr2m commented:
Yes, since the tests are only unit tests any way, which could end up preventing us from removing dead code in the future

laumair commented:
@gr2m @mastermatt Done! I have removed the unit tests.
